cassandra/cluster.py

```diff
@@ -1748,6 +1748,21 @@ def _prepare_all_queries(self, host):
                         if not success:
                             log.debug("Got unexpected response when preparing "
                                       "statement on host %s: %r", host, response)
+            """
```

Comment:
This is my opinion where we get the biggest improvement, we don't have to set the keyspace of the connection, we can just send the query with the keyspace parameter, not sure how to add it though but a big `if ProtocolVersion.uses_keyspace_flag(self._protocol_version)... else`